### ISSUE TITLE
Fastlane shell error while using git_commit

### ISSUE THREAD

**Issue by OfirKariv** [fastlane/fastlane#11960]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Hello, 
I'm facing some issues while trying to add, commit and push new files to git. 
This is a short version of my code:

```
lane :Version_Upload do
... #testing code lined
...
git_add(path: ".")
UI.message "Please enter commit message"
commitMess = gets
git_commit(path:".", message: commitMess.to_s)
push_to_git_remote( local_branch: branchName)
end

```

At this point I am facing to main issues:

1) git_add does not add any files to be commited.
While trying to test my code, I've put a breakpoint right after the "git_add" command, and typed "git status" in my terminal.
At that point I still saw my changes in red color, and only after I manually typed "git add ." in the terminal I saw the files in green color.

2) **and this is the main issue** - 
I'm simply not able to commit with git_commit function.
After the code reaches git_commit line i'm getting the following error:

> Exit status of command 'git commit -m <Message i've typed> .' was 1 instead of 0.
> On branch <My Branch Name>
> Changes not staged for commit:
> <list of files that has been modified>
> no changes added to commit.

At this point fastlane crush. I'm getting the fastlane summery is displayed and at the end of it I get the following error:

> Exit status of command 'git commit -m <Message i've typed> .' was 1 instead of 0. (FastlaneCore::Interface::FastlaneShellError)

> from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane_core/lib/fastlane_core/ui/ui.rb:16:in `method_missing'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/helper/sh_helper.rb:80:in `sh_control_output'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/helper/sh_helper.rb:12:in `sh'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/actions/git_commit.rb:11:in `run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:231:in `chdir'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
> 	from Fastfile:153:in `block (2 levels) in parsing_binding'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/lane.rb:33:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/lane.rb:33:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/lane_manager.rb:57:in `cruise_lane'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/commands_generator.rb:337:in `run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/cli_tools_distributor.rb:98:in `take_off'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/bin/fastlane:20:in `<top (required)>'

I've been trying to change the "path" and the "message" parameter in git_commit and still couldn't make it works.
What am I missing?

Thank you

Update:
fastlane env - 

## Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.3                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.3 (17D47)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

\\\\\\\\\\\\\
<my code>
\\\\\\\\\\\\\\

</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.83.0  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.6        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| commander-fastlane        | 4.4.6        |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| gh_inspector              | 1.1.2        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| highline                  | 1.7.10       |
| excon                     | 0.60.0       |
| CFPropertyList            | 2.3.6        |
| libxml-ruby               | 3.0.0        |
| plist                     | 3.4.0        |
| fastimage                 | 2.1.1        |
</details>

**Comment by OfirKariv** [comment]
Updated

**Comment by OfirKariv** [comment]
Problem solved!
Apparently, the issue was that I located the fastlane directory in the wrong place. 
The fastlane directory should be in your git local repo (and not in a subfolder of it) if you want to use git commands via fastlane.
